routes/orders.js

```diff
@@ -1,55 +1,49 @@
-const { response } = require('express');
-
 const ordersRouter = require('express').Router();
-// const { getAllProducts, postProduct} = require('../db/products');
-// const { getAllOrders, getPendingOrderByUser } = require('../db/orders');
-ordersRouter.get("/", async (req, res, next) => {
-    try {
-    // const products = await getAllProducts();
-    // const orders = await getAllOrders();
-    console.log("products:", products)
-    res.send({
-      orders: orders
-    });
-  } catch (error) {
-  }
-});
-ordersRouter.post("/", async (req, res, next) => {
-    const {status, userId, datePlaced } = req.body
-  try {
-    const newProduct = await postProduct(status = created, userId, datePlaced);
-    // const orders = await getAllOrders();
-    console.log("newlyAddedProduct:", newProduct)
-    req.user ? res.send({
-      newProduct : newProduct
-    }) : null
-  } catch (error) {
-  }
+const { getAllProducts } = require('../db/products');
+const { getAllOrders, getCartByUser, createOrder } = require('../db/orders');
+const { requireUser, requireAdmin } = require('./utils');
+
+ordersRouter.get('/', requireAdmin, async (req, res, next) => {
+	try {
+		const orders = await getAllOrders();
+		res.send({ orders });
+	} catch ({ name, message }) {
+		next({ name, message });
+	}
 });
-ordersRouter.get("/cart", async (req, res, next) => {
-try {
-  const cartToCheckout = await getPendingOrderByUser(req.user);
-  // const orders = await getAllOrders();
-  console.log("newlyAddedProduct:", newProduct)
-  req.user ? res.send({
-    cartToCheckout : cartToCheckout
-  }) : null
-} catch (error) {
-}
+
+ordersRouter.get('/cart', requireUser, async (req, res, next) => {
+	try {
+		const cartOrders = await getCartByUser(req.user);
+		res.send({ cartOrders });
+	} catch (error) {}
 });
-ordersRouter.get("/users/:userId/orders", async (req, res, next) => {
-  const { userId } = req.params
-  try {
-    const cartToCheckout = await getPendingOrdersByUser(userId);
-    // const orders = await getAllOrders();
-    console.log("newlyAddedProduct:", newProduct)
-    req.user ? res.send({
-      cartToCheckout : cartToCheckout
-    }) : null
-  } catch (error) {
-  }
-  });
 
+ordersRouter.post('/', async (req, res, next) => {
+	const { status } = req.body;
+	const datePlaced = Date.now();
 
+	try {
+		const newOrder = await createOrder(status, req.user.id, datePlaced);
+
+		res.send({ newOrder });
+	} catch ({ name, message }) {
+		next({ name, message });
+	}
+});
+
+ordersRouter.get('/users/:userId/orders', async (req, res, next) => {
+	const { userId } = req.params;
+	try {
+		const cartToCheckout = await getPendingOrdersByUser(userId);
+		// const orders = await getAllOrders();
+		console.log('newlyAddedProduct:', newProduct);
```

Comment:
I am not sure what this console.log is doing but i need to edit these changes next so i will merge in and test.